src/rgw/rgw_op.cc

```diff
@@ -5035,13 +5035,11 @@ void RGWPutACLs::execute()
     if (op_ret < 0)
       return;
 
-    new_data = strdup(ss.str().c_str());
-    free(data);
-    data = new_data;
-    len = ss.str().size();
+    data.clear();
+    data.append(ss.str().c_str(), ss.str().size());
```

Comment:
calling ss.str() twice will make two copies of the output string. there's a std::string overload for append, so this could just be `data.append(ss.str())`